pool/web/view/layout/navigation_experiment.ml

```diff
@@ -21,30 +21,47 @@ let title_to_string language text =
   | Text str -> str
 ;;
 
-let nav_elements { Experiment.id; _ } =
+let nav_elements { Experiment.id; direct_registration_disabled; _ } =
   let open Guard in
   let open ValidationSet in
   let open I18n in
   let target_id = id |> Uuid.target_of Experiment.Id.value in
-  [ "", Overview, Experiment.Guard.Access.read id
-  ; ( "assistants"
-    , Field Field.Assistants
-    , Or
-        [ SpecificRole `ManageAssistants
-        ; SpecificRole (`ManageAssistant target_id)
-        ] )
-  ; ( "experimenter"
-    , Field Field.Experimenter
-    , Or
-        [ SpecificRole `ManageExperimenters
-        ; SpecificRole (`ManageExperimenter target_id)
-        ] )
-  ; "invitations", Invitations, Invitation.Guard.Access.index id
-  ; "waiting-list", WaitingList, Waiting_list.Guard.Access.index id
-  ; "sessions", Sessions, Session.Guard.Access.index id
-  ; "assignments", Assignments, Assignment.Guard.Access.index id
-  ; "mailings", Mailings, Mailing.Guard.Access.index id
-  ]
+  let left =
+    [ "", Overview, Experiment.Guard.Access.read id
+    ; ( "assistants"
+      , Field Field.Assistants
+      , Or
+          [ SpecificRole `ManageAssistants
+          ; SpecificRole (`ManageAssistant target_id)
+          ] )
+    ; ( "experimenter"
+      , Field Field.Experimenter
+      , Or
+          [ SpecificRole `ManageExperimenters
+          ; SpecificRole (`ManageExperimenter target_id)
+          ] )
+    ; "invitations", Invitations, Invitation.Guard.Access.index id
+    ]
+  in
+  let right =
+    [ "sessions", Sessions, Session.Guard.Access.index id
+    ; "assignments", Assignments, Assignment.Guard.Access.index id
+    ; "mailings", Mailings, Mailing.Guard.Access.index id
+    ]
+  in
+  let nav_links =
```

Comment:
Please update it to something like this:

```ocaml
let waiting_list_nav =
  if Experiment.(DirectRegistrationDisabled.value direct_registration_disabled)
  then [ "waiting-list", WaitingList, Waiting_list.Guard.Access.index id ]
  else []
in
left @ waiting_list_nav @ right

Reply:
@mabiede done